include/model.php

```diff
@@ -702,34 +702,26 @@ public function filter_language_terms_orderby( $orderby, $args, $taxonomies ) {
 	 * @phpstan-return list<PLL_Language>
 	 */
 	protected function get_languages_from_taxonomies() {
-		$terms_by_type = $this->get_language_terms_by_type();
 
-		if ( empty( $terms_by_type['post'] ) || empty( $terms_by_type['term'] ) ) {
-			// `post` and `term` languages are mandatory.
-			return array();
-		}
+		// To get language taxonomies first.
+		$reversed_terms = array_reverse( $this->get_language_terms() );
 
-		$languages = array();
+		$terms_by_slug = array();
 
-		foreach ( $terms_by_type['post'] as $term_slug => $lang_term ) {
-			if ( ! isset( $terms_by_type['term'][ "pll_{$term_slug}" ] ) ) {
-				// A corresponding `term` language is mandatory.
-				continue;
+		foreach ( $reversed_terms as $term ) {
+			if ( $term instanceof WP_Term ) {
+				// Except for language taxonomy term slugs, remove 'pll_' prefix from the other language taxonomy term slugs.
+				$key = 'language' === $term->taxonomy ? $term->slug : substr( $term->slug, 4 );
+				$terms_by_slug[ $key ][ $term->taxonomy ] = $term;
 			}
+		}
 
-			$lang_terms = array();
-
-			foreach ( $terms_by_type as $type => $terms ) {
-				$key = 'post' === $type ? $term_slug : "pll_{$term_slug}";
+		// To put the language in the right order.
+		$terms_by_slug = array_reverse( $terms_by_slug );
 
-				if ( isset( $terms[ $key ] ) ) {
-					$lang_terms[ $type ] = $terms[ $key ];
-				}
-			}
+		$languages = array();
 
-			if ( ! isset( $lang_terms['post'], $lang_terms['term'] ) ) {
-				continue;
-			}
+		foreach ( $terms_by_slug as $lang_terms ) {
```

Comment:
I suggest to simply the loop because the factory always returns an object.
```php
foreach ( $terms_by_slug as $lang_terms ) {
	$languages[] = PLL_Language_Factory::get_from_terms( $lang_terms );
}
```
```

Reply:
Done in 6f3885e36e5950089a072d17901cf88f15275d96